src/components/LoanCards/KvClassicLoanCardContainer.vue

```diff
@@ -274,7 +285,68 @@ export default {
 		},
 		setCookie(name, value, options) {
 			return this.cookieStore.set(name, value, options);
-		}
+		},
+		async fetchLoanActivity() {
+			const activities = [];
+
+			const activityResponse = await this.apollo.query({
+				query: loanActivitiesQuery,
+				variables: { loanId: this.loanId }
+			});
+
+			const lendingActionsData = activityResponse.data?.lend?.loan?.lendingActions?.values ?? [];
+			const commentsData = activityResponse.data?.lend?.loan?.comments?.values ?? [];
+
+			lendingActionsData.forEach(action => {
+				const actionDate = new Date(action?.latestSharePurchaseDate).toDateString();
+				if (!activities.some(activity => activity.key === actionDate)) {
+					activities.push({
+						key: actionDate,
+						data: [],
+					});
+				}
+				if (action?.lender.name) {
```

Comment:
Is optional chaining needed here? `action?.lender?.name`

Reply:
There is no activity w/o lender data, but good practice.